### PR TITLE
INTER-4172: Completing the Order Twice is Causing Locking in Meta V2

### PR DESCRIPTION
**Pull request by pavel-bystritskyi**

### CODE REVIEW

**Review by mwraich — COMMENTED** [review]
Couple minor nits and suggestions, but tested on V2 and it looks like it's working in the view but the state call isn't updating the platformID as it should in the checkout DB.

Not sure why this is happening, tried to dig into the logs but looks like gconsole is down. Will dig more into it tomorrow
Screenshot 2024-07-14 at 1 59 34 PM

Screenshot 2024-07-14 at 1 48 38 PM


OrderID: `KVuntA9ZFS5kiMhvYeouz7Qimofv7w35dtWr6mJqfCw4lrUmRX8gtky0SL4Fb3TC`
https://github.com/user-attachments/assets/921fe1f6-7c17-4c86-9332-ddea27efb54e

File: Model/Order/CompleteOrderPool.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        );
+        $flowType = $quoteExtensionData->getFlowType();
+        $processor = $this->pool[$flowType] ?? null;
+        if ($processor instanceof CompleteOrderInterface) {
+            $processor->execute($order);
+        } else {
+            $this->logger->error(
+                __('Failed to find complete processor for order with id="%1"', $order->getEntityId())
+            );
+        }
```
Comment (JosephLeedy) [inline Model/Order/CompleteOrderPool.php:78]: \[Nitpick\]: I would invert this conditional to eliminate the `else` statement for improved readability.

File: Model/Order/CompleteOrderPool.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
+<?php
```
[the remark this answers is not in the document]
Comment (JosephLeedy) [reply Model/Order/CompleteOrderPool.php:1]: It looks like author of this PR is attempting to support multiple methods of completing an order. I'm unsure why that functionality is necessary, though.

[the remark this answers is not in the document]
Comment (JosephLeedy) [reply Model/Order/CompleteOrderPool.php:1]: Ah, reviewing further, they have two separate pieces of logic: one for the original complete order functionality and another that uses the new Simple Order endpoints.

File: Model/Order/PlaceOrder.php [hunk 3]
```diff
@@ -437,24 +435,6 @@ static function (array $transaction): bool {
 
         $this->updateCheckoutSession($quote, $order);
 
-        try {
```
Comment (JosephLeedy) [inline Model/Order/PlaceOrder.php:440]: Please verify that removing this logic doesn't break the corresponding tests.

File: etc/db_schema.xml [hunk 4]
```diff
@@ -26,6 +26,7 @@
         <column xsi:type="int" name="id" unsigned="true" nullable="false" identity="true" comment="ID"/>
         <column xsi:type="int" name="quote_id" unsigned="true" nullable="false" identity="false" comment="Magento Quote ID"/>
         <column xsi:type="smallint" name="order_created" default="0" comment="Magento Order Should Be Created On Magento Side"/>
+        <column xsi:type="varchar" name="flow_type" nullable="true" comment="Bold Flow Type"/>
```
Comment (JosephLeedy) [inline etc/db_schema.xml:29]: Should this column have a size defined, or do we just want to use the default of 255?

**Review by NickolasMalovanets — APPROVED** [review]

**Pull request merged**